From 30827be4f38704fc56b35fab61d60abe69976c04 Mon Sep 17 00:00:00 2001
From: prasannas <prasannas@ami.com>
Date: Tue, 14 Nov 2023 17:36:53 +0530
Subject: [PATCH 139/139] Fix For Network IPMI Policy in Redfish

Description:Disable And Enable Outband of ipmi in Redfish
1)Enable Outband of IPMI In redfish
Patch call
https://x.x.x.x/redfish/v1/Managers/bmc/NetworkProtocol

Request:
{
    "IPMI": {
        "ProtocolEnabled": true
    }
}

Response:
204 No Content

2)Disable Outband of IPMI In redfish
Patch call
https://x.x.x.x/redfish/v1/Managers/bmc/NetworkProtocol

Request:
{
    "IPMI": {
        "ProtocolEnabled": false
    }
}

Response:
204 No Content

---
 redfish-core/lib/network_protocol.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/redfish-core/lib/network_protocol.hpp b/redfish-core/lib/network_protocol.hpp
index de51166..84c3ca6 100644
--- a/redfish-core/lib/network_protocol.hpp
+++ b/redfish-core/lib/network_protocol.hpp
@@ -577,7 +577,7 @@ inline void handleManagersNetworkProtocolPatch(
     {
         handleProtocolEnabled(
             *ipmiEnabled, asyncResp,
-            encodeServiceObjectPath(std::string(ipmiServiceName) + '@'));
+            encodeServiceObjectPath(std::string(ipmiServiceName)));
     }
 
     if (sshEnabled)
-- 
2.9.5

